src/Components/Web/src/Routing/NavigationLock.cs

```diff
@@ -9,13 +9,12 @@ namespace Microsoft.AspNetCore.Components.Routing;
 /// <summary>
 /// A component that can be used to intercept navigation events. 
 /// </summary>
-public sealed class NavigationLock : IComponent, IAsyncDisposable
+public sealed class NavigationLock : ComponentBase, IAsyncDisposable
```

Comment:
The change of base class is a public API change - not a super scary one but a real one nonetheless.

Is this because the complexity is higher if implementing `IHandleAfterRender` etc?

Reply:
Right, I did this to reduce the complexity of this class and reuse the existing logic in `ComponentBase`.

Do you think there are any disadvantages to changing the base class to `ComponentBase` rather than implementing `IHandleAfterRender` (which is also technically a public API change)? This type is already `sealed` so there's no risk of breaking derived types, and it still implements `IComponent` via `ComponentBase`. I could be missing something but to me it seems like both options have similar significance.

Reply:
I agree it's not much of a difference, so if you think it's more maintainable on `ComponentBase` that's fine with me.

Where possible I've tried to build the baked-in components on `IComponent` as a way of minimizing API surface and to retain strong proof that Blazor doesn't accidentally start relying on *all* components being built on `ComponentBase`. However it's not worth doing that if `NavigationLock` ends up being harder to maintain that way.

Reply:
> Where possible I've tried to build the baked-in components on `IComponent` as a way of minimizing API surface and to retain strong proof that Blazor doesn't accidentally start relying on all components being built on `ComponentBase`.

That seems fair, and I anticipate the complexity increase of implementing `IHandleAfterRender` manually should be insignificant. I'll give that approach a shot.